crates/ironrdp-cliprdr/src/lib.rs

```diff
@@ -70,29 +100,29 @@ impl Cliprdr {
 
     fn handle_server_capabilities(&mut self, server_capabilities: Capabilities) -> PduResult<Vec<SvcMessage>> {
         self.capabilities.downgrade(&server_capabilities);
+        self.backend
+            .on_receive_downgraded_capabilities(self.capabilities.flags());
 
         // Do not send anything, wait for monitor ready pdu
         Ok(vec![])
     }
 
     fn handle_monitor_ready(&mut self) -> PduResult<Vec<SvcMessage>> {
-        let response = [
-            ClipboardPdu::Capabilities(self.capabilities.clone()),
-            ClipboardPdu::TemporaryDirectory(ClientTemporaryDirectory::new(self.temporary_directory.clone())?),
-            ClipboardPdu::FormatList(self.build_local_format_list()?),
-        ]
-        .into_iter()
-        .map(into_cliprdr_message)
-        .collect();
-
-        Ok(response)
+        // Request client to sent list of initially available formats and wait for the backend
+        // response.
+        self.backend.on_request_format_list();
```

Comment:
I had to go on a very long trip over two channels to hunt down what this ultimately does, and it looks like it eventually calls back into this `Cliprdr` object to get it to send back what we need:

https://github.com/Devolutions/IronRDP/blob/8a21b4d982031498b051ca3a973bb2ff78836a1c/crates/ironrdp-cliprdr/src/lib.rs#L180-L193

I think we can simplify this by changing this from `on_request_format_list` to something like `get_available_formats`

```rust
/// Returns the vector of available formats in the local clipboard.
fn get_available_formats(&self) -> Vec<ClipboardFormat>;
```

Then here we can do

```rust
fn handle_monitor_ready(&mut self) -> PduResult<Vec<SvcMessage>> {
    let temporary_directory = self.backend.temporary_directory();
    let available_formats = self.backend.get_available_formats();

    let response = [
        ClipboardPdu::Capabilities(self.capabilities.clone()),
        ClipboardPdu::TemporaryDirectory(ClientTemporaryDirectory::new(temporary_directory)?),
        ClipboardPdu::FormatList(self.build_format_list(&available_formats).unwrap()),
    ]
    .into_iter()
    .map(into_cliprdr_message)
    .collect::<Vec<_>>();

    Ok(response)
}
```